_episodes/debug.md

```diff
@@ -10,16 +10,389 @@ objectives:
 keypoints:
 - "First key point. Brief Answer to questions. (FIXME)"
 ---
-By the end of this episode,
-learners should be able to
-__recognize and fix simple bugs in their workflow code__.
 
-(non-exhaustive) list of possible examples:
+> ## Learning objectives
+>
+> By the end of this episode, learners should be able to
+> __recognize and fix simple bugs in their workflow code__.
+{: .callout}
 
-- YAML errors
-- "wiring errors" e.g. where is the output from my step?
-- type mismatch
-- array vs single-item mismatch
-- no formats on input but format is required by workflow
+When working on a CWL workflow, you will probably encounter errors. There are many different errors possible.
+It is always very important to check the error message in the terminal, because it will give you information on the error.
+This error message will give you the type of error as well as the line of code that contains the error.
+Some of these errors will be explained in this episode.
 
+As a first step to check if your CWL script contains any errors, you can run the workflow with the `--validate` flag.
+~~~
+cwltool --validate CWL_SCRIPT.cwl
+~~~
+{: .language-bash}
+
+It is possible that the script is validated, however, it still gets an error.
+If you encounter an error, the best practice is to run the workflow with the `--debug` flag.
+This will provide you with extensive information on the error you encounter.
+~~~
+cwltool --debug CWL_SCRIPT.cwl
+~~~
+{: .language-bash}
+
+### YAML errors
+First of all, errors in the YAML syntax. When writing a piece of code, it is very easy to make a mistake.
+
+Some very common YAML errors are:
+
+- Using tabs instead of spaces. In YAML files indentations are made using spaces, not tabs.
+  Errors caused by tabs will show `'NoneType' object has no attribute 'name'`.
+
+  ~~~
+  cwlVersion: v1.2
+  class: Workflow
+
```

Comment:
Maybe a shorter example? This is currently longer than a screenful with not much advantage for that

Reply:
There's a tradeoff between using the example from the rest of the lesson that learners will be familiar with vs using a 'hello world' example which would be shorter but unrelated. What do you think?